lib/hooks/useFormDelayContext.tsx

```diff
@@ -0,0 +1,116 @@
+import { logMessage } from "@lib/logger";
+import { FormElement, FormProperties } from "@lib/types";
+import { createContext, useContext, useState } from "react";
+
+interface FormDelay {
+  startTime: number;
+  requiredQuestions: number;
+}
+
+const timerDefault: FormDelay = {
+  startTime: 0,
+  requiredQuestions: 0,
+};
+
+const FormDelayContext = createContext<{
+  formDelay: FormDelay;
+  setFormDelay: React.Dispatch<React.SetStateAction<FormDelay>>;
+}>({
+  formDelay: timerDefault,
+  setFormDelay: () => {},
+});
+
+export const FormDelayProvider = ({ children }: { children: React.ReactNode }) => {
+  const [formDelay, setFormDelay] = useState(timerDefault);
+  return (
+    <FormDelayContext.Provider value={{ formDelay, setFormDelay }}>
+      {children}
+    </FormDelayContext.Provider>
+  );
+};
+
+// Adds in a little math to make it more unpredictable and calculates the final form delay
+const calculateSubmitDelay = (delayFromFormData: number) => {
+  const secondsBaseDelay = 2; // To help test group forms, make this big e.g. 20000
+  const secondsPerFormElement = 2;
+  return secondsBaseDelay + delayFromFormData * secondsPerFormElement;
+};
+
+export const calculateDelayWithGroups = (
+  startTime: number,
+  endTime: number,
+  requiredQuestions: number
+) => {
+  if (isNaN(startTime) || isNaN(endTime) || isNaN(requiredQuestions)) {
+    return -1;
+  }
+  const elapsedTime = Math.floor((endTime - startTime) / 1000);
+  const delayFromFormData = requiredQuestions - elapsedTime;
+  return calculateSubmitDelay(delayFromFormData);
+};
+
+export const calculateDelayWithoutGroups = (formElements: FormElement[]) => {
+  if (!Array.isArray(formElements)) {
+    return -1;
+  }
+  const delayFromFormData = formElements.filter(
+    (element) => element.properties.validation?.required === true
+  ).length;
+  return calculateSubmitDelay(delayFromFormData);
+};
+
+export const useFormDelay = () => {
+  const { formDelay, setFormDelay } = useContext(FormDelayContext);
```

Comment:
It's usually good practice to check if the Context is available from where the hook is being called.  If the context doesn't exist, the calling node is not a child node of the context then we throw an error.  https://github.com/cds-snc/platform-forms-client/blob/8cdf199ef04cf38b10f5898c9e9a3d9a9ac3c2d8/app/(gcforms)/%5Blocale%5D/(form%20administration)/form-builder/%5Bid%5D/edit/logic/components/flow/provider/FlowRefProvider.tsx#L30